backend/alembic/versions/9c615d167588_add_tasks.py

```diff
@@ -0,0 +1,62 @@
+"""Add Tasks
+
+Revision ID: 9c615d167588
+Revises: 617bf951f6a2
+Create Date: 2018-07-14 19:07:16.544507
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+# revision identifiers, used by Alembic.
+revision = '9c615d167588'
+down_revision = '617bf951f6a2'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table('Tasks',
+                    sa.Column('_created', sa.DateTime(), server_default=sa.text('now()'), nullable=False),
+                    sa.Column('_modified', sa.DateTime(), server_default=sa.text('now()'), nullable=False),
+                    sa.Column('id', sa.Integer(), autoincrement=True, nullable=False),
+                    sa.Column('key', sa.String(length=50), nullable=False),
+                    sa.Column('name', sa.String(length=100), nullable=False),
+                    sa.Column('image_path', sa.String(length=100), nullable=False),
+                    sa.PrimaryKeyConstraint('id', name=op.f('pk_Tasks')),
+                    sa.UniqueConstraint('key', name=op.f('uq_Tasks_key'))
+                    )
+    op.create_table('ScanCategories_Tasks',
+                    sa.Column('scan_category_id', sa.Integer(), nullable=True),
+                    sa.Column('task_id', sa.Integer(), nullable=True),
+                    sa.ForeignKeyConstraint(['scan_category_id'], ['ScanCategories.id'],
+                                            name=op.f('fk_ScanCategories_Tasks_scan_category_id_ScanCategories')),
+                    sa.ForeignKeyConstraint(['task_id'], ['Tasks.id'],
+                                            name=op.f('fk_ScanCategories_Tasks_task_id_Tasks'))
+                    )
+    op.add_column('LabelTags', sa.Column('task_id', sa.Integer(), nullable=True))
+    op.drop_constraint('fk_LabelTags_scan_category_id_ScanCategories', 'LabelTags', type_='foreignkey')
+    op.create_foreign_key(op.f('fk_LabelTags_task_id_Tasks'), 'LabelTags', 'Tasks', ['task_id'], ['id'])
+    op.drop_column('LabelTags', 'scan_category_id')
+    op.add_column('Labels', sa.Column('task_id', sa.Integer(), nullable=True))
+    op.create_foreign_key(op.f('fk_Labels_task_id_Tasks'), 'Labels', 'Tasks', ['task_id'], ['id'])
+
+    # Now Labels and LabelTags must refer to Task, so old records should be removed
```

Comment:
I've got some mixed feelings about it... We should avoid migrations which will drop user's data, especially Labels! I know that it will be a challenge to prepare a proper migration that will handle your change ;) It won't be easy... I think it may be impossible to do it properly...

(Proposal) Maybe it would be fine to start over with Alembic migrations? Maybe we should remove all of the versions and prepare a single new migration that will create whole database based on an empty DB? This is not so stupid as our SQL model has changed a lot since the beginning. What's more, we are still not production ready and doesn't even have a live experiment right now. Maybe it's high time to do something like this?

Reply:
For upgrade I will create a new Task for each ScanCategory (no data lost), for downgrade I won't do this better.